File: next/security/user and permission/backend-ms1/src/main/java/ma/zs/stocky/ws/facade/admin/security/RoleUserRestAdmin.java
package  ma.zs.stocky.ws.facade.admin.security;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.media.Content;
import io.swagger.v3.oas.annotations.media.Schema;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.responses.ApiResponses;

import ma.zs.stocky.bean.core.security.RoleUser;
import ma.zs.stocky.dao.criteria.core.security.RoleUserCriteria;
import ma.zs.stocky.service.facade.admin.security.RoleUserAdminService;
import ma.zs.stocky.ws.converter.security.RoleUserConverter;
import ma.zs.stocky.ws.dto.security.RoleUserDto;
import ma.zs.stocky.zynerator.controller.AbstractController;
import ma.zs.stocky.zynerator.dto.AuditEntityDto;
import ma.zs.stocky.zynerator.util.PaginatedList;


import org.springframework.core.io.InputStreamResource;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;


import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import ma.zs.stocky.zynerator.process.Result;


import org.springframework.web.multipart.MultipartFile;
import ma.zs.stocky.zynerator.dto.FileTempDto;

@RestController
@RequestMapping("/api/admin/roleUser/")
public class RoleUserRestAdmin  extends AbstractController<RoleUser, RoleUserDto, RoleUserCriteria, RoleUserAdminService, RoleUserConverter> {



    @Operation(summary = "upload one roleUser")
    @RequestMapping(value = "upload", method = RequestMethod.POST, consumes = "multipart/form-data")
    public ResponseEntity<FileTempDto> uploadFileAndGetChecksum(@RequestBody MultipartFile file) throws Exception {
        return super.uploadFileAndGetChecksum(file);
    }
    @Operation(summary = "upload multiple roleUsers")
    @RequestMapping(value = "upload-multiple", method = RequestMethod.POST, consumes = "multipart/form-data")
    public ResponseEntity<List<FileTempDto>> uploadMultipleFileAndGetChecksum(@RequestBody MultipartFile[] files) throws Exception {
        return super.uploadMultipleFileAndGetChecksum(files);
    }

    @Operation(summary = "Finds a list of all roleUsers")
    @GetMapping("")
    public ResponseEntity<List<RoleUserDto>> findAll() throws Exception {
        return super.findAll();
    }



    @Operation(summary = "Saves the specified  roleUser")
    @PostMapping("")
    public ResponseEntity<RoleUserDto> save(@RequestBody RoleUserDto dto) throws Exception {
        return super.save(dto);
    }

    @Operation(summary = "Updates the specified  roleUser")
    @PutMapping("")
    public ResponseEntity<RoleUserDto> update(@RequestBody RoleUserDto dto) throws Exception {
        return super.update(dto);
    }

    @Operation(summary = "Delete list of roleUser")
    @PostMapping("multiple")
    public ResponseEntity<List<RoleUserDto>> delete(@RequestBody List<RoleUserDto> listToDelete) throws Exception {
        return super.delete(listToDelete);
    }
    @Operation(summary = "Delete the specified roleUser")
    @DeleteMapping("")
    public ResponseEntity<RoleUserDto> delete(@RequestBody RoleUserDto dto) throws Exception {
            return super.delete(dto);
    }

    @Operation(summary = "Delete the specified roleUser")
    @DeleteMapping("id/{id}")
    public ResponseEntity<Long> deleteById(@PathVariable Long id) throws Exception {
        return super.deleteById(id);
    }
    @Operation(summary = "Delete multiple roleUsers by ids")
    @DeleteMapping("multiple/id")
    public ResponseEntity<List<Long>> deleteByIdIn(@RequestBody List<Long> ids) throws Exception {
            return super.deleteByIdIn(ids);
     }


    @Operation(summary = "find by user id")
    @GetMapping("user/id/{id}")
    public List<RoleUserDto> findByUserId(@PathVariable Long id){
        return findDtos(service.findByUserId(id));
    }
    @Operation(summary = "delete by user id")
    @DeleteMapping("user/id/{id}")
    public int deleteByUserId(@PathVariable Long id){
        return service.deleteByUserId(id);
    }
    @Operation(summary = "find by role id")
    @GetMapping("role/id/{id}")
    public List<RoleUserDto> findByRoleId(@PathVariable Long id){
        return findDtos(service.findByRoleId(id));
    }
    @Operation(summary = "delete by role id")
    @DeleteMapping("role/id/{id}")
    public int deleteByRoleId(@PathVariable Long id){
        return service.deleteByRoleId(id);
    }

    @Operation(summary = "Finds a roleUser and associated list by id")
    @GetMapping("id/{id}")
    public ResponseEntity<RoleUserDto> findById(@PathVariable Long id) {
        return super.findById(id);
    }

    @Operation(summary = "Finds roleUsers by criteria")
    @PostMapping("find-by-criteria")
    public ResponseEntity<List<RoleUserDto>> findByCriteria(@RequestBody RoleUserCriteria criteria) throws Exception {
        return super.findByCriteria(criteria);
    }

    @Operation(summary = "Finds paginated roleUsers by criteria")
    @PostMapping("find-paginated-by-criteria")
    public ResponseEntity<PaginatedList> findPaginatedByCriteria(@RequestBody RoleUserCriteria criteria) throws Exception {
        return super.findPaginatedByCriteria(criteria);
    }

    @Operation(summary = "Exports roleUsers by criteria")
    @PostMapping("export")
    public ResponseEntity<InputStreamResource> export(@RequestBody RoleUserCriteria criteria) throws Exception {
        return super.export(criteria);
    }

    @Operation(summary = "Gets roleUser data size by criteria")
    @PostMapping("data-size-by-criteria")
    public ResponseEntity<Integer> getDataSize(@RequestBody RoleUserCriteria criteria) throws Exception {
        return super.getDataSize(criteria);
    }



    public RoleUserRestAdmin (RoleUserAdminService service, RoleUserConverter converter) {
        super(service, converter);
    }




}

File: nest/security-v2/backend-ms1/src/app/module/ws/admin/security/RoleUserRestAdmin.ts
import {Body, Controller,Delete , Get, Param, Post, Put} from '@nestjs/common';
import {ApiOperation, ApiTags} from '@nestjs/swagger';
import {PaginatedList} from "src/app/zynerator/util/PaginatedList";

import {RoleUserConverter} from "src/app/shared/converter/security/RoleUserConverter";
import {RoleUserDto} from "src/app/shared/dto/security/RoleUserDto";
import {RoleUserCriteria} from "src/app/shared/dao/criteria/core/security/RoleUserCriteria";
import {RoleUserAdminServiceImpl} from "src/app/module/service/admin/security/RoleUserAdminServiceImpl";


@ApiTags('Manages role user services')
@Controller('api/admin/roleUser')
export class RoleUserRestAdmin {

    constructor(private readonly service: RoleUserAdminServiceImpl,
                private readonly converter: RoleUserConverter) {
    }

    @ApiOperation({summary: 'Finds a list of all roleUsers'})
    @Get()
    async findAll(): Promise<RoleUserDto[]> {
        const items = await this.service.findAll();
        return this.converter.toDtos(items);
    }

    @ApiOperation({summary: 'Finds a role user by id'})
    @Get('id/:id')
    async findById(@Param('id') id: number): Promise<RoleUserDto> {
        const item = await this.service.findById(id);
        return this.converter.toDto(item);
    }

    @ApiOperation({summary: 'Deletes a role user by id'})
    @Delete('id/:id')
    async deleteById(@Param('id') id: number): Promise<number> {
        return this.service.deleteById(id);
    }

    @ApiOperation({summary: 'Saves the specified role user'})
    @Post()
    async save(@Body() dto: RoleUserDto): Promise<RoleUserDto> {
        const item = this.converter.toItem(dto);
        const savedItem = await this.service.save(item);
        return this.converter.toDto(savedItem);
    }


    @ApiOperation({summary: 'Updates the specified role user'})
    @Put()
    async update(@Body() dto: RoleUserDto): Promise<RoleUserDto> {
        const item = this.converter.toItem(dto);
        const result = await this.service.updateWithAssociatedLists(item);
        return this.converter.toDto(result);
   }

    @ApiOperation({summary: 'Finds an optimized list of all roleUsers'})
    @Get('optimized')
    async findAllOptimized(): Promise<RoleUserDto[]> {
        const result = await this.service.findAllOptimized();
        return result;
    }

    @ApiOperation({summary: 'Finds an optimized list of all roleUsers'})
    @Post('find-by-criteria')
    async findByCriteria(@Body() criteria: RoleUserCriteria): Promise<RoleUserDto[]> {
        const items = await this.service.findByCriteria(criteria);
        return this.converter.toDtos(items);
    }

    @ApiOperation({summary: 'Finds an optimized list of all roleUsers'})
    @Post('find-paginated-by-criteria')
    async findPaginatedByCriteria(@Body() criteria: RoleUserCriteria): Promise<PaginatedList<RoleUserDto>> {
        const paginated = await this.service.findPaginatedByCriteria(criteria);
        const dtos = this.converter.toDtos(paginated.list);
        return new PaginatedList<RoleUserDto>(dtos, paginated.dataSize);
    }


    @Get('user/id/:id')
    async findByUserId(@Param('id') id: number): Promise<RoleUserDto[]> {
        const items = await this.service.findByUserId(id);
        return this.converter.toDtos(items);
    }
    @Get('role/id/:id')
    async findByRoleId(@Param('id') id: number): Promise<RoleUserDto[]> {
        const items = await this.service.findByRoleId(id);
        return this.converter.toDtos(items);
    }

    @Post('multiple')
    async deleteMultiple(@Body() dtos: RoleUserDto[]): Promise<RoleUserDto[]> {
        const items = dtos.map(dto => this.converter.toItem(dto));
        const deletedItems = await this.service.deleteMultiple(items);
        return deletedItems.map(element => this.converter.toDto(element));
    }

}
